File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/ui/preferences/locationathan/numeric/NumericDialog.java
package com.byagowi.persiancalendar.ui.preferences.locationathan.numeric;

import android.os.Build;
import android.text.InputType;
import android.view.View;
import android.widget.EditText;

import androidx.preference.EditTextPreferenceDialogFragmentCompat;

/**
 * Created by ebraminio on 2/21/16.
 */
public class NumericDialog extends EditTextPreferenceDialogFragmentCompat {

    @Override
    protected void onBindDialogView(View view) {
        super.onBindDialogView(view);

        EditText editText = view.findViewById(android.R.id.edit);
        editText.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_FLAG_SIGNED |
                InputType.TYPE_NUMBER_FLAG_DECIMAL);

        // on platforms supporting direction as LTR direction is more handy on editing numbers
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
            editText.setTextDirection(View.TEXT_DIRECTION_LTR);
            editText.setLayoutDirection(View.LAYOUT_DIRECTION_LTR);
        }
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/ui/preferences/locationathan/numeric/NumericDialog.kt
package com.byagowi.persiancalendar.ui.preferences.locationathan.numeric

import android.text.InputType
import android.view.View
import android.widget.EditText
import androidx.preference.EditTextPreferenceDialogFragmentCompat

class NumericDialog : EditTextPreferenceDialogFragmentCompat() {
    override fun onBindDialogView(view: View?) {
        super.onBindDialogView(view)

        view?.findViewById<EditText>(android.R.id.edit)?.apply {
            inputType = InputType.TYPE_CLASS_NUMBER or InputType.TYPE_NUMBER_FLAG_SIGNED or
                    InputType.TYPE_NUMBER_FLAG_DECIMAL
            // LTR direction is more handy on editing numbers
            textDirection = View.TEXT_DIRECTION_LTR
            layoutDirection = View.LAYOUT_DIRECTION_LTR
        }
    }
}
